docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/launcher/DockerComputerSSHLauncher.java

```diff
@@ -150,7 +153,11 @@ protected InetSocketAddress getAddressForSSHD(String cloudId, InspectContainerRe
              * like Joyent's Triton. */
                 if (host == null || host.equals("0.0.0.0") || usesSingleHostAbstraction(ir)) {
                     // Try to connect to the container directly (without going through the host)
-                    host = networkSettings.getIpAddress();
+                    if (network != null && network.length() > 0) {
+                        host = networkSettings.getNetworks().get(network).getIpAddress();
```

Comment:
In this code path, the slave is not connected to via the exposed port, but directly via port 22 (cf. https://github.com/amerkel2/docker-plugin/blob/edd1a87f32ec216d573b0c5cf63b299b2607d8ba/docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/launcher/DockerComputerSSHLauncher.java#L161). The objective of this PR is to improve detection of the virtual network IP of the container. That the container cannot be reached via this IP in all scenarios (e.g., if the master runs on a different physical host) is a different matter that could indeed be solved by adding the host to connect to via a separate config option.